client/api/request.js

```diff
@@ -42,8 +19,13 @@ Object.defineProperty(client, 'base', {
   }
 });
 
+client.setStorageInterface = function ({ getToken, clearAuthData }) {
+  this._getAuthToken = getToken;
+  this._clearAuthData = clearAuthData;
+};
```

Comment:
Previously `Auth` class reported back to `vuex` and now we need bidirectional communication between `vuex/persist` and `request` instance. Token flows from `vuex` to `request` and token invalidation requests go opposite way. Ofc that means we can't use events anymore and next step would be implementing request-reply pattern by using specially crafted pairs of request and reply events but that is too much work for this simple use case so we ended up with alternative solution that is basically passing interfaces around. That is fine but maybe we can make it even better - by better I mean more decoupled - by doing something along these lines.
From `request`'s perspective we can use local `request.token` variable and just get it and set it to `null` when needed (invalidation due to server response). From `vuex`'s perspective we can _decorate_ `request` with `token` property:
```js
Object.defineProperty(request, 'token', {
  get: () => store.state.auth.token,
  set: value => !value && store.dispatch('logout')
});
```
As far as `request` is concerned it is changing and reading local variable while vuex intercepts that calls and acts accordingly.